### PR TITLE
Add config files for IntelliJ IDEA.

### PR DESCRIPTION
**Pull request by scooby**
As with the Pycharm PR, I'm guessing that Jetbrains will release 2017.2 and 2017.3, so it seems reasonable to simply add those entries now.

There's an open issue to add wildcards, I think that would be inappropriate as it could lead to some pretty horrific breakage and it's not much of a lift to update this every year or so.

Let me know if this doesn't work for you, I can restrict it to only adding 2017.1.

### CODE REVIEW

File: mackup/applications/intellijidea.cfg [hunk 1]
```diff
@@ -15,10 +15,21 @@ Library/Preferences/IntelliJIdea14
...
-Library/Preferences/IdeaIC2016.1
 Library/Preferences/IntelliJIdea2016.2
 Library/Application Support/IntelliJIdea2016.2
 Library/Preferences/IntelliJIdea2016.3
 Library/Application Support/IntelliJIdea2016.3
+Library/Preferences/IdeaIC2017.1
+Library/Preferences/IdeaIC2017.2
+Library/Preferences/IdeaIC2017.3
+Library/Preferences/IntelliJIdea2017.1
+Library/Application Support/IdeaIC2017.1
```
Comment (kaliberr44) [inline mackup/applications/intellijidea.cfg]: I do have:
Library/Application Support/IntelliJIdea2017.1 
not 
Idea2017.1

Comment (scooby) [inline mackup/applications/intellijidea.cfg]: Are you running community edition? Because I am, and am showing "Idea2017.1" in App Support.

Comment (kaliberr44) [inline mackup/applications/intellijidea.cfg]: Ultimate

Comment (kaliberr44) [inline mackup/applications/intellijidea.cfg]: Could you also add this path: 
Library/Application Support/IntelliJIdea2017.1 

It's for Ultimate Edition

**Comment by lcardito** [comment]
This has been open quite a while. Can it be merged?

File: mackup/applications/intellijidea.cfg (hunk 1 above)
Comment (scooby) [reply mackup/applications/intellijidea.cfg]: Just popped that in.

**Comment by scooby** [comment]
I've merged from master, no conflicts I could see.

**Comment by MadeOfMagicAndWires** [comment]
As there was a unresolved conflict in CHANGELOG, I took the liberty of also adding @wicksome to the credits, as #1047 implements the same issue. Thought it'd be nice to give both of you credit but if you don't agree feel free to roll-back .

**Comment by MadeOfMagicAndWires** [comment]
Similarly to #1013, I've added @wangkezun to the credits.

**Comment by alienlydimmish** [comment]
Hi there,  looks like this has been open for a while — could this PR be merged? Would be very helpful to backup IntelliJ settings

Happy new year:)

**Comment by lra** [comment]
@alienlydimmish the problem with those half baked PRs is that it takes time to finish them, and the upstream is even gone.
The most efficient way to contribute is to deal with comments and keep the PR up to date with master.
I took it over in https://github.com/lra/mackup/pull/1675

**Pull request closed without merging**